src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/OrderItemsController.cs

```diff
@@ -98,24 +98,36 @@ public async Task<ActionResult<IEnumerable<GetOrderItemModel>>> GetAllAsync(stri
                 orderItems = orderItems.Where(y => y.CatalogueItemType.Equals(catalogueItemTypeFromName));
             }
 
-            var serviceRecipients = order.ServiceRecipients;
+            var serviceRecipientDictionary = order.ServiceRecipients.ToDictionary(x => x.OdsCode.ToUpperInvariant());
 
             return orderItems
                 .OrderBy(x => x.Created)
-                .Select(orderItem => new GetOrderItemModel
-                {
-                    ItemId = $"{orderId}-{orderItem.OdsCode}-{orderItem.OrderItemId}",
-                    ServiceRecipient = new ServiceRecipientModel
-                    {
-                        Name = serviceRecipients.FirstOrDefault(serviceRecipient => string.Equals(orderItem.OdsCode,
-                            serviceRecipient.OdsCode, StringComparison.OrdinalIgnoreCase))?.Name,
-                        OdsCode = orderItem.OdsCode
-                    },
-                    CataloguePriceType = orderItem.CataloguePriceType.Name,
-                    CatalogueItemType = orderItem.CatalogueItemType.Name,
-                    CatalogueItemName = orderItem.CatalogueItemName,
-                    CatalogueItemId = orderItem.CatalogueItemId,
-                }).ToList();
+                .Select(orderItem => new GetOrderItemModel(orderItem, serviceRecipientDictionary[orderItem.OdsCode.ToUpperInvariant()])).ToList();
+        }
+
+        [HttpGet("{orderItemId}")]
```

Comment:
Will it not be better to match the other routes, and have a seperate [Route] containing the orderItemId parameter rather than adding it to the get?

Reply:
Yes that makes sense. I will add the route attribute.